addons/project/models/project.py

```diff
@@ -1290,9 +1281,9 @@ def default_get(self, default_fields):
 
     def read(self, fields=None, load='_classic_read'):
         if fields and self.env.user.has_group('base.group_portal'):
-            for key in fields:
-                if key not in self.SELF_READABLE_FIELDS:
-                    raise AccessError(_('You cannot read %s field in task.') % key)
+            unauthorized_fields = set(fields) - self.SELF_READABLE_FIELDS
+            if unauthorized_fields:
+                raise AccessError(_('You cannot read %s field in task.', ', '.join(unauthorized_fields)))
```

Comment:
Maybe this could be moved to a utility method taking a list of fields? Because this exact pattern seems to be used half a dozen times.

Reply:
Indeed, I will do this.